server/mailet/integration-testing/src/test/java/org/apache/james/smtp/SmtpRandomStoringTest.java

```diff
@@ -130,30 +142,43 @@ private void createJamesServer() throws Exception {
             .build(temporaryFolder.newFolder());
     }
 
-    @After
-    public void tearDown() {
+    private static void sendMails() throws Exception {
+        SMTPMessageSender authenticatedSmtpConnection = messageSender.connect(LOCALHOST_IP, jamesServer.getProbe(SmtpGuiceProbe.class).getSmtpPort())
+            .authenticate(FROM, PASSWORD);
+
+        IntStream.range(0, NUMBER_OF_MAILS)
+            .forEach(Throwing.intConsumer(index ->
+                authenticatedSmtpConnection
+                    .sendMessage(buildMail("Message " + index))).sneakyThrow());
+    }
+
+    @AfterClass
+    public static void tearDown() {
         jamesServer.shutdown();
     }
 
     @Test
-    public void sendingOneHundredMessagesShouldBeRandomlyAssignedToEveryMailboxesOfEveryUsers() throws Exception {
-        int numberOfMails = 100;
+    public void oneHundredMailsShouldHaveBeenStoredBetweenFourAndEightTimes() {
```

Comment:
are you making tests depend on each other?

Reply:
no, they all depends of the provisioning, done at class level.